src/ristretto/pedersen/extended_commitment_factory.rs

```diff
@@ -0,0 +1,415 @@
+        let k_zero = vec![RistrettoSecretKey::default(); ExtensionDegree::Five as usize];
+        for extension_degree in EXTENSION_DEGREE {
+            let mut v_sum = RistrettoSecretKey::default();
+            let mut k_sum = vec![RistrettoSecretKey::default(); extension_degree as usize];
+            let factory = ExtendedPedersenCommitmentFactory::new_with_extension_degree(extension_degree).unwrap();
+            let mut c_sum = factory.commit(&k_zero[0..extension_degree as usize], &v_zero).unwrap();
+            let mut commitments = Vec::with_capacity(25);
+            for _ in 0..25 {
+                let v = RistrettoSecretKey::random(&mut rng);
+                v_sum = &v_sum + &v;
+                let k_i = vec![RistrettoSecretKey::random(&mut rng); extension_degree as usize];
+                for i in 0..extension_degree as usize {
+                    k_sum[i] = &k_sum[i] + &k_i[i];
+                }
+                let c = factory.commit(&k_i, &v).unwrap();
+                c_sum = &c_sum + &c;
+                commitments.push(c);
+            }
+            assert!(factory.open(&k_sum, &v_sum, &c_sum).unwrap());
+            assert_eq!(c_sum, commitments.iter().sum());
+        }
+    }
+
+    #[test]
+    fn serialize_deserialize() {
+        let mut rng = rand::thread_rng();
+        for extension_degree in EXTENSION_DEGREE {
+            let factory = ExtendedPedersenCommitmentFactory::new_with_extension_degree(extension_degree).unwrap();
+            let k = vec![RistrettoSecretKey::random(&mut rng); extension_degree as usize];
+            let c = factory.commit_value(&k, 420).unwrap();
+            // Base64
+            let ser_c = c.to_base64().unwrap();
+            let c2 = PedersenCommitment::from_base64(&ser_c).unwrap();
+            assert!(factory.open_value(&k, 420, &c2).unwrap());
+            // MessagePack
+            let ser_c = c.to_binary().unwrap();
+            let c2 = PedersenCommitment::from_binary(&ser_c).unwrap();
+            assert!(factory.open_value(&k, 420, &c2).unwrap());
+            // Invalid Base64
+            assert!(PedersenCommitment::from_base64("bad@ser$").is_err());
+        }
+    }
+
+    #[test]
+    fn derived_methods() {
+        for extension_degree in EXTENSION_DEGREE {
+            let factory = ExtendedPedersenCommitmentFactory::new_with_extension_degree(extension_degree).unwrap();
+            let k = vec![RistrettoSecretKey::from(1024); extension_degree as usize];
+            let c1 = factory.commit_value(&k, 2048).unwrap();
+            let mut hasher = DefaultHasher::new();
+            c1.hash(&mut hasher);
+            let c2 = c1.clone();
+            let c3 = factory.commit_value(&k, 2049).unwrap();
+
+            // Test 'clone`
+            assert_eq!(c1, c2);
+            assert_ne!(c2, c3);
+
+            // Note! 'Ord' and 'PartialOrd' are undefined for extended commitments with extension degree other than Zero
```

Comment:
Why undefined? Point ordering is lexicographic on canonical byte representations, and makes sense for any group element. (That being said, ordering has no algebraic significance whatsoever...)

Reply:
I think what I should have said is that for this specific test _"When manipulating v and k_i, ordering of the commitment result is unpredictable for extended commitments with extension degree other than Zero, but certainly repeatable"_. Anyways, I fixed the test to be robust with changing extension degrees